firestore-shorten-urls-dynamic-links/functions/src/index.ts

```diff
@@ -100,7 +100,12 @@ class FirestoreDynamicLinksUrlShortener extends FirestoreUrlShortener {
         method: "POST",
         body: JSON.stringify(requestBody),
       });
-      const { shortLink: shortUrl } = await response.json();
+      const responseBody = await response.json();
+      if (!response.ok) {
+        this.logs.error(responseBody.error.message);
```

Comment:
Small nit; bring the `response.json()` call after the ok check. If the payload is large, it's spending time in the function for no reason.

Reply:
The `response.json()` value is used in both ways.

If response not ok:
```js
this.logs.error(responseBody.error.message);
```

if it is ok:
```js
const shortUrl = responseBody.shortLink;
```